ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -877,14 +893,13 @@ public HttpContent readChunk(ByteBufAllocator allocator) throws Exception {
       Chunk chunk = chunksToWrite.poll();
       if (chunk != null) {
         chunk.onDequeue();
-        ByteBuf buf = Unpooled.wrappedBuffer(chunk.buffer);
-        progress.addAndGet(chunk.buffer.remaining());
+        progress.addAndGet(chunk.buffer.readableBytes());
         chunksAwaitingCallback.add(chunk);
         if (chunk.isLast) {
-          content = new DefaultLastHttpContent(buf);
+          content = new DefaultLastHttpContent(chunk.buffer);
```

Comment:
should these buffers be retained before sending as content? If the chunk callback called by `operationProgressed` releases the buffer and the content object is also released by downstream channel handlers (not sure if this part happens or not), it may get an IllegalReferenceCountException

Reply:
Confirmed that the outgoing ByteBuf will be released once it reaches the end of the pipeline: https://netty.io/wiki/reference-counted-objects.html#outbound-messages